### ISSUE TITLE
Resolution of richtext elements

### ISSUE THREAD

**Issue by CaarLo1337** [e-Spirit/javascript-content-api-library#213]
There are some problems with resolving the RichText

When creating lists using RichText, a `<p>` element is erroneously displayed before the `<ul>` element. This results in invalid HTML structure and causes errors in Next.js.

Current Workaround
To avoid these errors, we are currently rendering all paragraph elements as `<div>`. While this prevents errors, it also removes `<p>` elements entirely from the RichText content.

```
{
    "data": {},
    "content": [
        {
            "data": {
                "data-fs-style": "0",
                "data-fs-property-style": "0"
            },
            "content": [
                {
                    "data": {},
                    "content": [
                        {
                            "type": "text",
                            "content": "list",
                            "data": {}
                        }
                    ],
                    "type": "listitem"
                },
                {
                    "data": {},
                    "content": [
                        {
                            "type": "text",
                            "content": "list",
                            "data": {}
                        }
                    ],
                    "type": "listitem"
                }
            ],
            "type": "list"
        }
    ],
    "type": "paragraph"
}
```
![image](https://github.com/user-attachments/assets/513a49e7-e587-42b2-a00e-ee08f72336a7)

The same applies to the H1-H6 formats. Here, “Block” is played as the type although you would expect it to play as type “H1” “H2” etc. instead of via the data-style

![image](https://github.com/user-attachments/assets/c63b2f9b-e65d-4e71-b3df-02207a153214)

**Comment by jodeleit-es** [comment]
Hi CaarLo1337,
you are correct, in FirstSpirit lists (UL) are always nested in a paragraph (P). But it's important to note that neither in CaaS nor in the content API we are creating HTML. Both provide documents models which are defined in FirstSpirit and which not directly compatibly to the HTML document model. So it must be mapped with some effort to HTML. That the used names in FirstSpirit are HTML tag names does not make this discussion easier. 

To sum this up: your solution is a perfectly valid way to handle the transformation to HMTL.

**Comment by neo-reply-lukas** [comment]
To rectify the structure, you could split the content array on type=list, and pull the list item up a level
e.g.
```
const input = [
  {
    content: [text1, text2, list, text3],
    type: "paragraph",
  },
];
// gets transformed to
const transformed = [
  { content: [text1, text2], type: "paragraph" },
  {
    content: list.content,
    type: "list",
  },
  [{content: [text3], type: "paragraph"}]
];
```
